docs/INSTALL.md

```diff
@@ -87,84 +87,105 @@ recommended for mainnet. The `master` branch can at times be unstable and
-    repo can now live anywhere!
+In order to work with [`lnd`](https://github.com/lightningnetwork/lnd), the 
+following build dependencies are required:
+
+### Go
+
+`lnd` is written in Go, with a minimum version of 1.16. To install, run one of 
+the following commands for your OS:
+
+<details>
+  <summary>Linux (x86-64)</summary>
+  
+  ```
+  wget https://dl.google.com/go/go1.17.1.linux-amd64.tar.gz
+  sha256sum go1.17.1.linux-amd64.tar.gz | awk -F " " '{ print $1 }'
+  ```
+
+  The final output of the command above should be
+  `dab7d9c34361dc21ec237d584590d72500652e7c909bf082758fb63064fca0ef`. If it
+  isn't, then the target REPO HAS BEEN MODIFIED, and you shouldn't install
+  this version of Go. If it matches, then proceed to install Go:
+  ```
+  sudo tar -C /usr/local -xzf go1.17.1.linux-amd64.tar.gz
+  export PATH=$PATH:/usr/local/go/bin
+  ```
+</details>
+
+<details>
+  <summary>Linux (ARMv6)</summary>
+  
+  ```
+  wget https://dl.google.com/go/go1.17.1.linux-armv6l.tar.gz
+  sha256sum go1.17.1.linux-armv6l.tar.gz | awk -F " " '{ print $1 }'
+  ```
+
+  The final output of the command above should be
+  `ed3e4dbc9b80353f6482c441d65b51808290e94ff1d15d56da5f4a7be7353758`. If it
+  isn't, then the target REPO HAS BEEN MODIFIED, and you shouldn't install
+  this version of Go. If it matches, then proceed to install Go:
+  ```
+  tar -C /usr/local -xzf go1.17.1.linux-armv6l.tar.gz
+  export PATH=$PATH:/usr/local/go/bin
+  ```  
+  
+</details>
+
+<details>
+  <summary>macOS</summary>
+  
+  First, install [Homebrew](https://brew.sh) if you don‘t already have it.
+
+  Then
+
+  ```
+  brew install go
+  ```
+
+</details>
+
+<details>
+  <summary>FreeBSD</summary>
+  
+  ```
+  pkg install go
+  ```
+
+  Alternatively, one can download the pre-compiled binaries hosted on the
+  [Golang download page](https://golang.org/dl/). If one seeks to install
+  from source, then more detailed installation instructions can be found
+  [here](https://golang.org/doc/install).
+</details>
+
+--- 
+
+***Important***
+
+At this point, you should set your `$GOPATH` environment variable, which
+represents the path to your workspace. By default, `$GOPATH` is set to
+`~/go`. You will also need to add `$GOPATH/bin` to your `PATH`. This ensures
+that your shell will be able to detect the binaries you install.
+
+```shell
+⛰  export GOPATH=~/go
+⛰  export PATH=$PATH:$GOPATH/bin
+```
+
+--- 
+
+We recommend placing the above in your `.bashrc`, `.zshrc` or in a setup script 
+so that you can avoid typing this every time you open a new terminal window.
+
+
+### Go modules
+
+This project uses [Go modules](https://github.com/golang/go/wiki/Modules) 
+to manage dependencies as well as to provide *reproducible builds*.
+
+Usage of Go modules (with Go 1.13) means that you no longer need to clone
+`lnd` into your `$GOPATH` for development purposes. Instead, your `lnd`
+repo can now live anywhere!
```

Comment:
Perhaps add a note that for mobile development this is still required (because `gomobile` isn't updated I think).
Also, this section header needs to be updated in the TOC.

Reply:
Yes, good idea. Fixed.